action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -461,6 +472,18 @@ func (stateDB *StateDBAdapter) RevertToSnapshot(snapshot int) {
 		log.L().Error("Failed to get snapshot.", zap.Int("snapshot", snapshot))
 		return
 	}
+	// restore logs
+	if stateDB.revertLog {
+		stateDB.logs = stateDB.logs[:stateDB.logsSnapshot[snapshot]]
+		delete(stateDB.logsSnapshot, snapshot)
```

Comment:
why do we need `map` here? Is the distribution of sn sparse?
I think `slice` is enough

Reply:
updated in 3rd commit